tests/experiment/test_model.py

```diff
@@ -391,11 +392,37 @@ def test_repr_resolution_function(self):
         )
 
 
+def test_copy():
```

Comment:
you didn't overwrite `copy` in class Model - why the need to test it here?

Reply:
Moved to `test_topmost_nesting.py`.  These are test to ensure that the nested structure is functional by using a method from the topmost class